src/main/java/uk/gov/justice/hmpps/prison/service/BookingService.java

```diff
@@ -578,13 +583,38 @@ public void verifyBookingAccess(final Long bookingId, final String... rolesAllow
             agencyIds.addAll(Set.of("OUT", "TRN"));
         }
         if (agencyIds.isEmpty()) {
+            if (authenticationFacade.isClientOnly()) {
+                logClientUnauthorisedAccess(bookingId, rolesAllowed);
+            }
             throw EntityNotFoundException.withMessage("Offender booking with id %d not found.", bookingId);
         }
         if (!bookingRepository.verifyBookingAccess(bookingId, agencyIds)) {
+            logUserUnauthorisedAccess(bookingId, agencyIds, rolesAllowed);
             throw EntityNotFoundException.withMessage("Offender booking with id %d not found.", bookingId);
         }
     }
 
+    private void logClientUnauthorisedAccess(final Long bookingId, final String... rolesAllowed) {
+        final Map<String, String> logMap = new HashMap<>();
+        logMap.put("bookingId", bookingId.toString());
+        logMap.put("currentClientRoles", StringUtils.join(
+            authenticationFacade.getAuthentication() == null ? null : authenticationFacade.getAuthentication().getAuthorities(), ","));
+        logMap.put("rolesAllowed", StringUtils.join(rolesAllowed,","));
+        telemetryClient.trackEvent("ClientUnauthorisedBookingAccess", logMap, null);
+    }
+
+    private void logUserUnauthorisedAccess(final Long bookingId, final Set<String> agencyIds, final String... rolesAllowed) {
+        String bookingAgencyId = bookingRepository.getBookingAgency(bookingId).orElse(null);
+        final Map<String, String> logMap = new HashMap<>();
+        logMap.put("bookingId", bookingId.toString());
+        logMap.put("bookingCaseload", bookingAgencyId);
+        logMap.put("currentUser", authenticationFacade.getCurrentUsername());
+        logMap.put("currentUserRoles", StringUtils.join(
+            authenticationFacade.getAuthentication() == null ? null : authenticationFacade.getAuthentication().getAuthorities(), ","));
```

Comment:
all checks for authenticationFacade.getAuthentication() currently do a null check - so replicated this here to ensure logging doesn't cause an nullpointer